### PR TITLE
Update to rust-miniscript f365e236a1d2bbe2164f2e40d8b203ce9fc07879

### PR DESCRIPTION
**Pull request by sanket1729**
This is not the latest version, but a version before bitcoin 0.28.0-rc1. Until we have a secp-zkp release with secp 0.5.0, we can't update elements to use the latest rust-bitcoin 0.28.0-rc1 and this will be blocked.

### CODE REVIEW

**Comment by sanket1729** [comment]
@apoelstra, so this is failing **only the doctests** for MSRV 1.29 because of some linker error. I spent some time trying to figure out how to fix it. But I think given the latest push to move 1.41.1, I think we should just update MSRV. Note that rust-elements also have MSRV 1.36.

You can review all changes from 18f438e9c00acf770c414b58d235d1eb0db8c40c.

**Comment by apoelstra** [comment]
ACK moving rust-miniscript to 1.41.1.

**Comment by apoelstra** [comment]
utACK 3d54351496f919f5983c12799f5dd4b6fa105543

I didn't look as closely as I could've at this; but given the general pre-alpha state of this library I don't think it's worth trying to do things "properly" at this point. In future we should try using the secp256k1-zkp merge script, or better, the Elements<-Bitcoin Core merge script.

**Comment by sanket1729** [comment]
@apoelstra, thanks for the utACK. I plan to do some integration testing for this code because we advertise it. Hopefully, that will increase some confidence in the code.

File: src/miniscript/lex.rs [hunk 1]
```diff
@@ -147,16 +142,16 @@ impl TokenIter {
     }
 }
```
Comment (sanket1729) [inline src/miniscript/lex.rs:144]: Test

**Pull request merged**